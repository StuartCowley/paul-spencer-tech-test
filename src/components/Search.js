import React, { useState } from "react";
import "../styles/search.css"

const Search = () => {
  const [searchText, setSearchText] = useState("")

  return(
    <>
      <input className="search-box" type="text" onChange={(e) => setValue(e.target.value)}></input>
    </>
  )
}

export default Search;