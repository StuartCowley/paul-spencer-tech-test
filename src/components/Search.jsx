import React, { useState } from "react";
import PropTypes from "prop-types";

import Header from "./Header";
import getImages from "../requests/getImages";

import "../styles/search.css";

function Search({ setSearchResults, setInitialise }) {
  const [searchText, setSearchText] = useState("")

  const handleSubmit = async(event) => {
    event.preventDefault()
    setSearchResults(await getImages(searchText))
    setInitialise(true)
  }

  return(
    <div>
      <Header />
      <div>
        <form className="input-search" onSubmit={handleSubmit}>
          <input className="search-box" type="text" placeholder="Search..." onChange={(event) => setSearchText(event.target.value)}></input>
          <button className="search-button" type="submit">Launch</button>
        </form>
      </div>
    </div>
  )
}

Search.propTypes = {
  setSearchResults: PropTypes.func.isRequired,
}

export default Search;