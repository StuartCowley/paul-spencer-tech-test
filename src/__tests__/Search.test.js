import { getByRole, render, screen } from "@testing-library/react";
import Search from "../components/Search";

describe('Search Box', () => {
  const { asFragment } = render(<Search />)

  test("search box renders correctly", () => {
    expect(asFragment).toMatchSnapshot()
  })
})