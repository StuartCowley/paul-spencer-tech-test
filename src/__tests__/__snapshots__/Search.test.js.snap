// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Box search box renders correctly 1`] = `[Function]`;
